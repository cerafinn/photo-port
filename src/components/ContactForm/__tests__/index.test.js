import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ContactForm from '..';

afterEach(cleanup);

describe('Contact component', () => {
  // First Test
  it('renders', () => {
    render(<ContactForm />);
  });
  // Second Test
  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<ContactForm />);

    expect(asFragment()).toMatchSnapshot();
  })
})

describe('links are visible', () => {
  it('inserts text into the home link', () => {
    const { getByTestId } = render(<ContactForm />);

    expect(getByTestId('submit')).toHaveTextContent('Submit');
    expect(getByTestId('contact')).toHaveTextContent('Contact Me');
  });
})