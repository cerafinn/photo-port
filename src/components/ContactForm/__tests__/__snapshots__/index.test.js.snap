// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="contact"
    >
      Contact Me
    </h1>
    <form
      id="contact-form"
    >
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email Address:
        </label>
        <input
          name="email"
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          rows="5"
          type="message"
        />
      </div>
      <button
        data-testid="submit"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
